contracts/src/fp/OPSuccinctFaultDisputeGame.sol

```diff
@@ -0,0 +1,521 @@
+
+/// @title OPSuccinctFaultDisputeGame
+/// @notice An implementation of the `IFaultDisputeGame` interface.
+contract OPSuccinctFaultDisputeGame is Clone, ISemver {
+    ////////////////////////////////////////////////////////////////
+    //                         Enums                              //
+    ////////////////////////////////////////////////////////////////
+
+    enum ProposalStatus {
+        // The initial state of a new proposal
+        Unchallenged,
+        // A proposal that has been challenged but not yet proven
+        Challenged,
+        // An unchallenged proposal that has been proven valid with a verified proof
+        UnchallengedAndValidProofProvided,
+        // A challenged proposal that has been proven valid with a verified proof
+        ChallengedAndValidProofProvided,
+        // The final state after resolution, either GameStatus.CHALLENGER_WINS or GameStatus.DEFENDER_WINS.
+        Resolved
+    }
+
+    ////////////////////////////////////////////////////////////////
+    //                         Structs                            //
+    ////////////////////////////////////////////////////////////////
+
+    /// @notice The `ClaimData` struct represents the data associated with a Claim.
+    struct ClaimData {
+        uint32 parentIndex;
+        address counteredBy;
+        address prover;
+        Claim claim;
+        ProposalStatus status;
+        Timestamp deadline;
+    }
+
+    ////////////////////////////////////////////////////////////////
+    //                         Events                             //
+    ////////////////////////////////////////////////////////////////
+
+    /// @notice Emitted when the game is challenged.
+    /// @param challenger The address of the challenger.
+    event Challenged(address indexed challenger);
+
+    /// @notice Emitted when the game is proved.
+    /// @param prover The address of the prover.
+    event Proved(address indexed prover);
+
+    /// @notice Emitted when the game is resolved.
+    /// @param status The status of the game after resolution.
+    event Resolved(GameStatus indexed status);
+
+    ////////////////////////////////////////////////////////////////
+    //                         State Vars                         //
+    ////////////////////////////////////////////////////////////////
+
+    /// @notice The maximum duration allowed for a challenger to challenge a game.
+    Duration internal immutable MAX_CHALLENGE_DURATION;
+
+    /// @notice The maximum duration allowed for a proposer to prove against a challenge.
+    Duration internal immutable MAX_PROVE_DURATION;
+
+    /// @notice The game type ID.
+    GameType internal immutable GAME_TYPE;
+
+    /// @notice The dispute game factory.
+    IDisputeGameFactory internal immutable DISPUTE_GAME_FACTORY;
+
+    /// @notice The chain ID of the L2 network this contract argues about.
+    uint256 internal immutable L2_CHAIN_ID;
```

Comment:
Why do we need the `l2ChainId` if we have the `ROLLUP_CONFIG_HASH`, which contains that information?

Reply:
It's duplicative.